ra/ra.go

```diff
@@ -988,7 +1006,45 @@ func (ra *RegistrationAuthorityImpl) issueCertificate(
 		return emptyCert, err
 	}
 
-	if ra.publisher != nil {
+	if ra.ctpolicy != nil {
+		var ctCtx context.Context
+		var cancel func()
+		currentDeadline, ok := ctx.Deadline()
+		if !ok {
+			// Current context doesn't have a deadline, this should
+			// never happen so it's a internal server error... but
+			// we already issued the cert so we can't fail out now.
+			// Just use a background context with a 30s timeout added.
+			ctCtx, cancel = context.WithTimeout(context.Background(), time.Second*30)
+		} else {
+			// NOTE: We want to check how putting the SCT submission/collection
+			// affects calls to IssueCertificate so we take the current context
+			// and allocate 80% of the remaining time to calling CTPolicy.GetSCTs.
+			// This way if we exceed the child context we won't time out the
+			// parent call and can still return the cert to the user.
+			until := time.Until(currentDeadline)
+			ctCtx, cancel = context.WithTimeout(ctx, time.Duration(float64(until)*0.8))
+		}
+		defer cancel()
+		started := ra.clk.Now()
+		_, err := ra.ctpolicy.GetSCTs(ctCtx, cert.DER)
+		took := ra.clk.Since(started)
+		// The final cert has already been issued so actually return it to the
+		// user even if this fails since we aren't actually doing anything with
+		// the SCTs yet.
+		var state string
+		if err != nil {
+			fmt.Println("TEST", err)
+			state = "failure"
+			if err == context.DeadlineExceeded {
+				state = "deadlineExceeded"
+			}
+			ra.log.Info(fmt.Sprintf("ctpolicy.GetSCTs failed: %s", err))
```

Comment:
Is `Info` the right log level for this or should it be an audit error?

Reply:
Audit requires it be retained specially and inspected. Given we don't actually use this yet I think that isn't warrented, `Warning` might be better than `Info` though.